include/common/Driver.hpp

```diff
@@ -555,75 +554,27 @@ class Driver
 		}
 
 		/*Implementatoin of writing last state Y and YDot for each unit operation*/
-
-		for (int i = 0; i <= _sim->model()->maxUnitOperationId(); ++i)
+		std::ostringstream oss;
+		for (int i = 0; i < _writeLastStateUnitId.size(); ++i)
 		{
-			std::ostringstream oss;
+			const unsigned int lastStateUnitId = _writeLastStateUnitId[i];
```

Comment:
The unit operation ID does not have to be the index of the unit in `_models`! If the unit id happens to match the index in `_models`, it is only coincidentally.

Reply:
But in this case the contents of _writeLastStateId[i] contains the unit IDs for which last state is supposed to be saved. 
So isnt it already addressing the issue? 
Apologies if I am not able to understand the issue here

Reply:
Yes, it contains the *unit id*, but `_models` could look like this:
```
[unit_002, unit_000, unit_001]
```
That is, `models[0].unitOperationId() == 2` could happen.

Reply:
Ah okay. Thanks for the clarification.